README.md

```diff
@@ -179,26 +179,63 @@ Grype lets you define custom output formats, using [Go templates](https://golang
 
 - Specify the path to the template file (`-t ./path/to/custom.template`).
 
-- Grype's template processing uses the same data models as the `json` output format — so if you're wondering what data is available as you author a template, you can use the output from `grype <image> -o json` as a reference.
 
-**Example:** You could make Grype output data in CSV format by writing a Go template that renders CSV data and then running `grype <image> -o template -t ~/path/to/csv.tmpl`.
+**Example:** Suppose you want Grype to output specific data columns, in CSV format.
+
+(1) Grype's input template processing uses the same data models as found in the `json` output format Grype produces. So if you're wondering what data is available for you to use in a template, examine data objects found in the output file.
+
+```grype -o json SAMPLE.war  > SAMPLE.war.json```
+
+Looking inside `SAMPLE.war.json` you see a pattern of data objects similar to this:
+
+```json-detailed
+    "matches": [
+        {
+          "vulnerability"         : { },
+          "relatedVulnerabilities": [ ],
+          "matchDetails"          : [ ],
+          "artifact"              : { },
+        } ,  ...   // "matches": The array of objects, containing details for all vulnerabilities found ...
+        { }
+    ], 
+    "source"    : { },    
+    "distro"    : { },
+    "descriptor": { }
+```
+(2) Looking inside any of these JSON objects, you see nested values with Lower-case names. 
+To use these any of these values in the Golang template, first change the name of the JSON value you want to use to be Upper-case. 
+Also, use the `.` to drill into a nested object. With these observations you can use any data identifiers you choose to reference in your template files. 
+For example: The expression `{{.Vulnerability.Fix.Versions}}` relates to the JSON object, ` "vulnerability": { "fix": { "versions": [ ] ` .
+The expression `{{- range .Matches}}` says to iterate across the JSON array, ` "matches":` .
+More completely, here's what a custom `csv.tmpl` template file might contain:
 
-Here's what the `csv.tmpl` file might look like:
 ```gotemplate
-"Package","Version Installed","Vulnerability ID","Severity"
+"Artifact:","Severity:","Is Fixed:","Vulnerability:","Version:","Version Fixed:","Package URL:","Vulnerability Source:"
 {{- range .Matches}}
-"{{.Artifact.Name}}","{{.Artifact.Version}}","{{.Vulnerability.ID}}","{{.Vulnerability.Severity}}"
+"{{.Artifact.Name}}","{{.Vulnerability.Severity}}","{{.Vulnerability.Fix.State}}","{{.Vulnerability.ID}}","'{{.Artifact.Version}}","{{.Vulnerability.Fix.Versions}}","{{.Artifact.PURL}}","{{.Vulnerability.DataSource}}"
```

Comment:
This PR, I believe, stemmed from a request to simply add a character to the fix version of the normal table output when it's missing. Maybe we could just limit this PR to include the fix version in the CSV example? I think that would satisfy the requirement without a large change to document a feature that might be best explained more thoroughly?

Reply:
I like this solution. I'd much prefer to make the smallest change possible — and to be able to concretely articulate the benefit of that small change.

Reply:
Thank you. Please see the commit and comment, below.
As a newbie walking this path (many others to follow), I hope I've touched on important things to add clarity for those people. And get more/deeper usage out of the cool Grype.
* I've (attached) a diff PDF file, here
[ git  diff   README.md.pdf](https://github.com/anchore/grype/files/8005821/git.diff.README.md.pdf)
```README.md: 
(1) Slimmed down previous changes 
(2) Improved JSON data shape example with matches[] 
(3) Cite a useful example of a *double* dotted de-reference, in the template 
(4) Example now shows Docker and Warfile results, side by side. (Suggests wider utility for Grype).
```

Reply:
One can see the proposed README.md , rendered here:
https://github.com/freedom-isnotanarchy/grype/blob/doc-fix-enhancement-617-by-improving-templated-output-example/README.md